File: examples/governance/compliance_domain/lib/compliance_domain/compliance_review/review_condition.rb
module ComplianceDomain
  class ComplianceReview
    class ReviewCondition
      attr_reader :requirement, :met, :evidence

      def initialize(requirement:, met:, evidence:)
        @requirement = requirement
        @met = met
        @evidence = evidence
        check_invariants!
        freeze
      end

      def ==(other)
        other.is_a?(self.class) &&
          requirement == other.requirement &&
          met == other.met &&
          evidence == other.evidence
      end
      alias eql? ==

      def hash
        [self.class, requirement, met, evidence].hash
      end

      private

      def check_invariants!; end
    end
  end
end

File: examples/governance_static_go/domain/review_condition.go
package domain

type ReviewCondition struct {
	Requirement string `json:"requirement"`
	Met string `json:"met"`
	Evidence string `json:"evidence"`
}

func NewReviewCondition(requirement string, met string, evidence string) (ReviewCondition, error) {
	v := ReviewCondition{
		Requirement: requirement,
		Met: met,
		Evidence: evidence,
	}
	return v, nil
}
